src/OSPSuite.Core/Serialization/Xml/SerializationContext.cs

```diff
@@ -87,6 +87,10 @@ public DataColumnCollectionReference AddDataCollectionReference(ICache<Auxiliary
 
       public IDimension DimensionByName(string dimensionName)
       {
+         var found = DimensionFactory.TryGetDimension(dimensionName, out var dimension);
+         if (found)
+            return dimension;
+
          return DimensionFactory.Dimension(dimensionName);
       }
```

Comment:
If not found, same behavior as before (which is going to throw an exception)

Reply:
The last return statement `return DimensionFactory.Dimension(dimensionName);` will always throw an exception.
Why not 
```
if (found)
  return dimension;
else
 throw new XXXException

```?

Reply:
Because I don't want to know what exception to throw. If I leave the same code as before, nothing will change. Same exception, same message etc..